index.js

```diff
@@ -76,5 +76,12 @@ export default async function pRetry(input, options) {
 				}
 			}
 		});
+
+		if (options.signal && !options.signal.aborted) {
+			options.signal.addEventListener('abort', () => {
+				operation.stop();
+				reject(new Error(options.signal.reason ?? 'AbortError: This operation was aborted'));
```

Comment:
The fallback message is the same as the default message when `signal.reason` is supported but nothing is passed to `controller.abort()`.
That way we can use `reason` and still support platforms which don't have it yet.

Reply:
> Note: When abort() is called, the fetch() promise rejects with an Error of type DOMException, with name AbortError. - https://developer.mozilla.org/en-US/docs/Web/API/AbortController/abort

I think we should do the same.

It should also use this code when available; https://nodejs.org/api/errors.html#abort_err

Reply:
Are you sure we should wrap the `signal.reason` in an error? What if the `reason` is an error type already? I would check what Fetch does in this scenario.

Reply:
It looks like fetch discards the abort reason and just uses the default DOMException:

<img width="592" alt="Screen Shot 2" src="https://user-images.githubusercontent.com/1402241/156517438-f6994d91-b2c8-4f92-ba57-20143f276131.png">

Reply:
If we want to preserve the error:

```suggestion
				reject(options.signal.reason ?? new DOMException('blah blah'));
```

if we want to follow fetch:


```suggestion
				reject(new DOMException('blah blah'));
```

Reply:
We want to preserve the error.